src/app/item-popup/Socket.tsx

```diff
@@ -30,7 +29,10 @@ export default function Socket({
   // we should address whatever is changing plug order in DIM
   if (socket.craftingData) {
     plugOptions = [...plugOptions].sort(
-      compareBy((p) => socket.craftingData![p.plugDef.hash]?.requiredLevel ?? 0)
+      compareBy((p) =>
+        // shove retired perks to the bottom (our choice) and consider requiredLevel:undefined to be 0 (bungie data works this way)
+        p.cannotCurrentlyRoll ? 999 : socket.craftingData![p.plugDef.hash]?.requiredLevel ?? 0
+      )
```

Comment:
Can we move this to `buildDefinedSocket` so that the Records item popup perk list view benefits from it as well?

Should also include `unreliablePerkOption` with #9003 merged.

Reply:
can do once https://github.com/DestinyItemManager/DIM/pull/9007 is reviewed